src/main/java/tatracker/logic/commands/session/FilterModuleCommand.java

```diff
@@ -0,0 +1,51 @@
+package tatracker.logic.commands.session;
+
+import static java.util.Objects.requireNonNull;;
+import static tatracker.logic.parser.CliSyntax.PREFIX_MODULE;
+import tatracker.commons.core.Messages;
+import tatracker.logic.commands.Command;
+import tatracker.logic.commands.CommandResult;
+import tatracker.logic.commands.CommandWords;
+import tatracker.model.Model;
+import tatracker.model.session.DoneSessionPredicate;
+
+/**
+ * Filter Module based on user's inputs.
+ * Used under TSS view.
+ */
+public class FilterModuleCommand extends Command {
+
+        public static final String COMMAND_WORD = String.format("%s %s", CommandWords.MODULE, CommandWords.FILTER_MODEL);
+
+        public static final String MESSAGE_USAGE = COMMAND_WORD + ": Filters module."
+        + "Parameters: "
+        + "[" + PREFIX_MODULE + "MODULE] "
+        + "Example: " + COMMAND_WORD + " "
+        + PREFIX_MODULE + "CS2103T";
+
+        public static final String MESSAGE_SUCCESS = "Filtered Module List";
+        public static final String MESSAGE_INVALID_MODULE_CODE = "There is no sessions with the given module code.";
+
+        private final DoneSessionPredicate predicate;
+
+        public FilterModuleCommand(DoneSessionPredicate predicate) {
+                this.predicate = predicate;
+        }
+
+        @Override
+        public CommandResult execute(Model model) {
+                requireNonNull(model);
+                model.updateFilteredDoneSessionList(predicate);
```

Comment:
Is this command being used to change the doneSessionList? If so, you should refactor and change the name of the command and class. Otherwise, it won't be possible to understand it!